Debug App/Sources/View Controllers/New UI/KlarnaHeadless/MerchantHeadlessCheckoutKlarnaViewController+Klarna.swift

```diff
@@ -79,13 +79,67 @@ extension MerchantHeadlessCheckoutKlarnaViewController: PrimerHeadlessErrorableD
         }
     }
 
+    private func showResultsVC(checkoutData: PrimerCheckoutData?, error: Error?) {
+        // If the checkout flow was set for Manual Handling then we want to show the manualHandlingCheckoutData
+        if manualHandlingCheckoutData != nil {
+            presentResultsVC(checkoutData: manualHandlingCheckoutData, error: nil)
+            return
+        }
+        presentResultsVC(checkoutData: checkoutData, error: nil)
+    }
+    
     private func presentResultsVC(checkoutData: PrimerCheckoutData?, error: Error?) {
         let rvc = MerchantResultViewController.instantiate(checkoutData: checkoutData, error: error, logs: logs)
         navigationController?.popToRootViewController(animated: true)
         navigationController?.pushViewController(rvc, animated: true)
     }
 }
 
+extension MerchantHeadlessCheckoutKlarnaViewController: PrimerHeadlessUniversalCheckoutDelegate {
```

Comment:
Could `MerchantHeadlessCheckoutAvailablePaymentMethodsViewController` act as the delegate here? These methods are already defined there for other PMs

Reply:
Its an implementation detail since this is the merchant side code so not critical to avoid this duplication, but would be good to consolidate if possible